lib/src/commit.rs

```diff
@@ -93,15 +93,15 @@ impl Commit {
         &self.data.parents
     }
 
-    pub fn parents(&self) -> impl Iterator<Item = BackendResult<Commit>> + '_ {
+    pub fn parents(&self) -> impl Iterator<Item = BackendResult<Commit>> + use<'_> {
```

Comment:
IIUC, this will be the default with the 2024 edition, so I suppose we'll drop the `+ use<'_>` then. Is there still a reason to make the change now?

Reply:
I thought it would be inconsistent to leave `+ '_` here. My main motivation of this patch is to remove weird `'a: 'b, 'b` constraints.

Reply:
Sounds good. There's probably some Clippy lint that figures out that we can remove it whether it's `+ '_` or `+ use<'_>`.